pairinteraction/CMakeLists.txt

```diff
@@ -113,10 +113,77 @@ set_target_properties(wignerSymbols PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_link_libraries(pireal PRIVATE wignerSymbols)
 target_link_libraries(picomplex PRIVATE wignerSymbols)
 
-# Include Eigen
+set(MKL_ARCH "intel64" CACHE STRING "MKL architecture")
+set(MKL_LINK "dynamic" CACHE STRING "MKL linkage")
+set(MKL_THREADING "sequential" CACHE STRING "MKL threading backend")
+set(MKL_INTERFACE "lp64" CACHE STRING "MKL interface")
+find_package(MKL CONFIG)
+find_package(PkgConfig)
+if(MKL_FOUND)
+  target_link_libraries(pireal PUBLIC MKL::MKL)
+  target_link_libraries(picomplex PUBLIC MKL::MKL)
+  target_compile_definitions(pireal PUBLIC WITH_INTEL_MKL)
+  target_compile_definitions(picomplex PUBLIC WITH_INTEL_MKL)
+elseif(PKG_CONFIG_FOUND)
+  if(MKL_LINK STREQUAL "sdl")
+    pkg_check_modules(MKL mkl-sdl IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "sequential")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-seq IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "intel_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-iomp IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "gnu_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-gomp IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "tbb_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-tbb IMPORTED_TARGET)
+  endif()
+  if(MKL_FOUND)
+    target_link_libraries(pireal PUBLIC PkgConfig::MKL)
+    target_link_libraries(picomplex PUBLIC PkgConfig::MKL)
+    target_compile_definitions(pireal PUBLIC WITH_INTEL_MKL)
+    target_compile_definitions(picomplex PUBLIC WITH_INTEL_MKL)
+  endif()
+endif()
+
+find_package(BLAS)
+find_package(LAPACK)
```

Comment:
If I only install `libopenblas-dev` and not `liblapacke-dev`, this line still finds lapack: "-- Found LAPACK: /usr/lib/x86_64-linux-gnu/libopenblas.so". However, then, the program is of course not compiling properly. I used cmake version 3.22.1 with `cmake .. -DBLA_VENDOR=OpenBLAS`. Is there an option that we can use to prevent this line from finding lapack if only blas is installed?

Reply:
The problem is that some distributions do not bundle LAPACK**E** (note the trailing **E**, very important) as part of LAPACK (without E). CMake does not offer any bundled detection of LAPACK**E**, which is why there is the `FIXME` further down in the code. Having something that handles all the cases (bundled vs. split package, cross-platform aspects) is not easy. However, it might be a good idea to make it fail when LAPACK**E** is not available. I'll add something for that.